### PR TITLE
fix(gatsby-recipes): fix GatsbyShadowFile resource — don't error if package isn't installed

### PR DESCRIPTION
**Pull request by KyleAMathews**
Discovered this while testing #27721

- don't throw if read returns undefined (the NPMPackage might be installed)
- don't throw if the shadow file does not exist (same)
- return `dependsOn` for the theme's equivalent `NPMPackage` to ensure it's installed before we try to create the shadowed file

### CODE REVIEW

File: packages/gatsby-recipes/src/providers/gatsby/shadow-file.js
Comment (pvdz) [inline packages/gatsby-recipes/src/providers/gatsby/shadow-file.js:121]: I would emit the error as a `.verbose()` just in case it's something else.

**Review by pvdz — DISMISSED** [review]
Nit: this PR does three things, the title/desc don't cover that.
- don't throw if `read` returns `undefined`
- don't throw if the shadow file does not exist
- return `dependsOn` (unclear to me what this changes or fixes but recipes already uses this property so it's probably fine)

File: packages/gatsby-recipes/src/providers/gatsby/shadow-file.js
Comment (KyleAMathews) [reply packages/gatsby-recipes/src/providers/gatsby/shadow-file.js:121]: 👍

**Comment by KyleAMathews** [comment]
Updated description

**Review by pvdz — APPROVED** [review]

**Comment by pvdz** [comment]
I've accepted it but I think you can put the `.verbose()` log back as the CI seems to have failed over the utila package (which you removed afterwards), not the verbose log. I'd still be in favor of putting that .verbose back in before merging but either way I'll leave that up to you.

**Comment by KyleAMathews** [comment]
Oh? I didn't remove the `utila` package actually — just added it to the rollup.config.js and then removed it. @pieh confirmed offline that the reporter being added to packages like this causes trouble.

**Pull request merged**
